File: mergeSort.c
#include<stdio.h>

void merge(int a[],int p,int q,int r)
{
	int i,j,k,n1=q-p+1,n2=r-q;
	int L[n1],R[n2];
	for(i=0;i<n1;i++)
		L[i]=a[p+i];
	for(i=0;i<n2;i++)
		R[i]=a[q+i+1];
	i=0;
	j=0;
	k=p;
	while((i<n1)&&(j<n2))
	{
		if(L[i]<=R[i])
		{
			a[k]=L[i];
			i++;
		}
		else
		{
			a[k]=R[j];
			j++;
		}
		k++;
	}
	while(i<n1)
	{
		a[k]=L[i];
		k++;
		i++;
	}
	while(j<n2)
	{
		a[k]=R[j];
		k++;
		j++;
	}
}

void mergeSort(int a[],int p,int r)
{
	if(p<r)
	{
		int q=(p+r)/2;
	        mergeSort(a,p,q);
		mergeSort(a,q+1,r);
		merge(a,p,q,r);

	}
}

int main()
{
	int n,i;
	scanf("%d",&n);
	int a[n];
	for(i=0;i<n;i++)
		scanf("%d",&a[i]);
	mergeSort(a,0,n-1);
	for(i=0;i<n;i++)
		printf("%d ",a[i]);
	printf("\n");
}

File: mergeSort.py
def merge(A,p,q,r):
    n1 = q-p+1
    n2 = r-q
    L = []
    R = []
    for i in range(n1):
        L.append(A[p+i-1])
    for i in range(n2):
        R.append(A[q+i])
    i=0
    j=0
    k=p


    while i<n1 and j<n2:
        if L[i] <= R[j]:
            A[k]=L[i]
            i+=1
        else:
            A[k]=R[j]
            j+=1
        k+=1
    while i<n1:
        A[k]=L[i]
        k+=1
        i+=1
    while j<n2:
        A[k]=R[j]
        k+=1
        j+=1

def mergeSort(A,p,r):
    if p<r:
        q=int((p+r)/2)
        mergeSort(A,q+1,r)
        mergeSort(A,p,q)
        merge(A,p,q,r)


n = int(input("Enter length of the array : "))
arr = []
print("Enter array : ")
for i in range(n):
    a = int(input())
    arr.append(a)

print(arr)

mergeSort(arr,0,len(arr)-1)

print(arr)
